code-challenges.test.js

```diff
@@ -1,52 +1,287 @@
+    // Expected output: false
+    const hand4 = [7, 2, 7, 2, 7]
+    // Expected output: true
+    it("takes in an array of 5 numbers and determines whether or not the array is a full house", () => {
+        expect(fullHouse(hand1).toEqual(true))
+        expect(fullHouse(hand2).toEqual(false))
+        expect(fullHouse(hand3).toEqual(false))
+        expect(fullHouse(hand4).toEqual(true))
+    })
+})
+
+// /*
+
+//  FAIL  ./code-challenges.test.js
+//   encription
+//     ✓ take in a string and returns a string with a coded message (2 ms)
+//   containsLetter
+//     ✓ take in an array of words and a single letter and returns an array of all the words containing that particular letter
+//   fullHouse
+//     ✕ takes in an array of 5 numbers and determines whether or not the array is a full house
+
+//   ● fullHouse › takes in an array of 5 numbers and determines whether or not the array is a full house
+
+//     ReferenceError: fullHouse is not defined
+
+// */
+
+// ////////////////////////////////////////////////////////////////////////////////////////////////
+
+// // b) Create the function that makes the test pass.
+
+// /*
+
+// I didnt have time to post my Blockers on slack over the weekend, so I went online to find help solving this problem,  I made sure to thoroughly pseudocode so that I understand what was learned using the online resource
+
+// Pseudocode: 1.) Create a function called fullHouse
+// 2.) fullHouse will take in an array as a argument
+// 3.) Create a variable called uniqueNumbers
+// 4.) use spread operator to get a subset of the array argument.
+// 5.) use new set() method to get individual integers provided from the argument
+// 6.) use .map() method to iterate over uniqueNumbers and .filter() method. When .filter()'s value at index matches .map()'s value at index the value is filtered out.
+// 7.) use the .length method to evaluate the length of the returned array
+// 8.) use .includes() and a logical && to evaluate if count's length is 2 & 3 and return a boolean value
+
+// */
+// const hand1 = [5, 5, 5, 3, 3]
+//     // Expected output: true
+//     const hand2 = [5, 5, 3, 3, 4]
+//     // Expected output: false
+//     const hand3 = [5, 5, 5, 5, 4]
+//     // Expected output: false
+//     const hand4 = [7, 2, 7, 2, 7]
+//     // Expected output: true
+
+const fullHouse = (array) => {
+    const uniqueNumbers = [...new Set(array)]
+    const counts = uniqueNumbers.map(value1 => array.filter(value2 => value2 === value1).length)
+    return counts.includes(2) && counts.includes(3)
+}
+
+// console.log(fullHouse(hand1)) // output: true
+// console.log(fullHouse(hand2)) // output: false
+// console.log(fullHouse(hand3)) // output: false
+// console.log(fullHouse(hand4)) // output: true
+
+/*
+
+Im having a hard time looking for a yarn Matcher to get a passing test.  When I run function the return is true, false, false, true.  Theres an issue with me using the spread operator, on the argument.
```

Comment:
Remember that if you use the spread operator on the argument, it puts the argument in an array. So if the argument is an array, your overall argument is now that array inside another array, which will mess up your output if you don't account for that!